File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.java
class Solution {
    public int evalRPN(String[] tokens) {
        if(tokens.length == 0){
            return 0;
        }

        // Map <String, BinaryOperation<Integer, Integer>> map = Map.of(
        //     '+'
        // )

        Deque<Integer> stack = new ArrayDeque<>();

        for( int i = 0; i < tokens.length; i++){
            if(tokens[i].equals("+")){
                int b = stack.pop();
                int a = stack.pop();
                stack.push(a + b);
            } else if(tokens[i].equals("-")){
                int b = stack.pop();
                int a = stack.pop();
                stack.push( a - b);
            }else if(tokens[i].equals("*")){
                int b = stack.pop();
                int a = stack.pop();
                stack.push( b * a);
            }else if(tokens[i].equals("/")){
                int b = stack.pop();
                int a = stack.pop();
                stack.push( a / b);
            }else{
            stack.push(Integer.parseInt(tokens[i]));
            }
        }

        return stack.pop();
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-4.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = deque()

        for char in s:
            if char == '(' or char == '{' or char == '[':
                stack.append(char)
            else:
                if len(stack) == 0: return False
                top = stack.pop()
                if char == ')' and top != '(': return False
                if char == '}' and top != '{': return False
                if char == ']' and top != '[': return False

        
        return len(stack) == 0